### PR TITLE
Bug/1148 image assets working in production

### LINKED ISSUE #1148: Refactor image assets to work in deployed instances
Images working on heroku.

### Current Behavior

Images do not load as the hash added by rails asset pipeline does not propagate beyond filenames
### Possible Solution

1. Disable digest for image assets
2. Refactor JF image assets to use  ActionView::Helpers::AssetUrlHelper 

### Steps to Reproduce (for bugs)

if solution one
1. rake assets:precompile
2. observe images assets in `/public/frontend` have hashes, they should not
if solution 2
1. starting and serving app locally as production should work
2. app in heroku should also work

### Your Environment

* Version used: latest develop
* Browser Name and version: ff dev
* Operating System and version (desktop or mobile): desktop

### PR DESCRIPTION
**Pull request by AllenBW**
Phew. 

This pr refactors all jf image assets to use the digest manifest, ameliorates production image asset woes. 

### Description
This pr refactors all jf image assets to use the digest manifest, ameliorates production image asset woes. 

### Related Issue
closes #1148

### Motivation and Context
see #1148

### How Has This Been Tested?
App built for production, whole of app traversed while watching for 500's

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Adds database migrations (Requires all environments to run `rake db:migrate` to work)
- [ ] Adds new Gemfile dependencies (Requires all environments to run `bundle install` to work)
- [ ] Adds new ENV variables (description of ENV and purpose is required in the above description)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My code has been cleaned of commented, logging, and inline test code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/assets/javascripts/app/components/navigation/sidebar-item.html.erb [hunk 1]
```diff
@@ -3,7 +3,6 @@
      ui-sref="{{ ::vm.item.state }}"
      ui-sref-active="sidebar__item-body--active">
     <i ng-if="::vm.item.icon" ng-class="::vm.item.icon" class="fa fa-fw"></i>
-    <img ng-if="::vm.item.image" ng-src="::'image/' + vm.item.image + '.png'" alt="{{vm.item.label}} Icon"/>
```
Comment (stackus) [inline app/assets/javascripts/app/components/navigation/sidebar-item.html.erb]: How are sidebar icons displayed now?

File: app/assets/javascripts/app/components/product-description/product-description.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<div class="product-description">
+  <div ng-if="::!vm.product.img" class="product-description__image product-description__image--missing"></div>
+  <img ng-if="::vm.product.name == 'Centos 7'" class="product-description__image" src="<%= image_url 'products/centos.png' %>" alt="Product Icon"/>
```
Comment (stackus) [inline app/assets/javascripts/app/components/product-description/product-description.html.erb]: Move and use assets as static images, don't create a giant list of ng-ifs. This is not maintainable code.

Comment (AllenBW) [inline app/assets/javascripts/app/components/product-description/product-description.html.erb]: please clarify assets as static images

File: app/assets/javascripts/app/components/navigation/sidebar-item.html.erb (hunk 1 above)
Comment (AllenBW) [inline app/assets/javascripts/app/components/navigation/sidebar-item.html.erb]: this isn't used, `_app_urls.sass` holds/fetches sidebar icons

File: app/assets/javascripts/app/states/services/details/details.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<content-header short="true">
+  <div class="content-header__body">
+    <img ng-if="::vm.product.name == 'Centos 7'" class="service-details-heading__icon" src="<%= image_url 'products/centos.png' %>" alt="Product Icon"/>
```
Comment (stackus) [inline app/assets/javascripts/app/states/services/details/details.html.erb]: 1. You're triggering on user input, and specifically on something the system is designed to allow user to create. The very first moment a user creates a new product this code won't work for them.
2. Duplication of code. This exact list has been seen elsewhere. It's not maintainable
3. Massive conditional list. This is basically the same thing as a giant if-else-if conditional. It's not maintainable.

**Comment by stackus** [comment]
Adds too many $watches, adds code thats unmaintainable or won't work with any new content created by users.

File: app/assets/javascripts/app/components/product-description/product-description.html.erb (hunk 2 above)
Comment (stackus) [inline app/assets/javascripts/app/components/product-description/product-description.html.erb]: Static assets are assets that live under `public/`.

File: app/assets/javascripts/app/components/navigation/sidebar-item.html.erb (hunk 1 above)
Comment (stackus) [inline app/assets/javascripts/app/components/navigation/sidebar-item.html.erb]: You're telling me if I search for the items, I won't find anything that makes it possible to set an 'image' on those items?

Comment (AllenBW) [inline app/assets/javascripts/app/components/navigation/sidebar-item.html.erb]: yes. i will also tell you the a folder titled "/image" doesn't exist (in this branch or anywhere in master)

**Comment by AllenBW** [comment]
... you asked for it to work in a pinch, that we would fix it later... feedback on what

**Pull request closed without merging**